neonvm/controllers/virtualmachine_controller.go

```diff
@@ -1014,15 +1018,17 @@ func sshSecretSpec(virtualmachine *vmv1.VirtualMachine) (*corev1.Secret, error)
 
 // labelsForVirtualMachine returns the labels for selecting the resources
 // More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/
-func labelsForVirtualMachine(virtualmachine *vmv1.VirtualMachine) map[string]string {
+func labelsForVirtualMachine(virtualmachine *vmv1.VirtualMachine, runnerVersion *api.RunnerProtoVersion) map[string]string {
```

Comment:
Why does it have to be a pointer?

Reply:
Ah, because you check for nil

Reply:
There's other ways to do it (originally I was going to make it a bool and set the version within the function), but the reasoning is because there's two slightly different requirements:

1. in `podSpec`, when we're generating the pod to create, we want the labels to include the runner API version
2. in `updatePodMetadataIfNecessary`, we want the 'desired' labels to *not* include the runner API version, so that the existing ignoring logic respects that

the pointer solution is kind of annoying though, so I'm open to other ideas if you've got something :)

(the reason I didn't go with the bool approach is because boolean args often make readability hard)